src/error.rs

```diff
@@ -9,24 +8,28 @@ use std::{
     fmt::{Display, Formatter},
     io,
     path::PathBuf,
+    result,
 };
 /// Shorthand for a Result that returns an `OrcaError`.
-pub type Result<T> = anyhow::Result<T, OrcaError>;
-// pub type Result<T> = result::Result<T, OrcaError>;
+pub type Result<T> = result::Result<T, OrcaError>;
 
 /// Possible errors you may encounter.
 #[derive(Debug)]
 pub(crate) enum Kind {
     /// Returned if a file is not expected to exist.
     FileExists(PathBuf),
-    /// Returned if a file is expected to have a parent.
-    FileHasNoParent(PathBuf),
     /// Returned if an annotation was expected to exist.
     NoAnnotationFound(String, String, String),
+    /// Returned if a model save was attempted without an annotation set.
+    MissingAnnotationOnSave,
+    /// Returned if an annotation delete was attempted on a model's last annotation.
+    DeletingLastAnnotation(String, String, String),
```

Comment:
Don't we want to support annotationless models?

Reply:
Yes, we should but my question is whether `save_model` needs to support saving a model w/o annotation?

It came up while I was implementing this so thought I'd offer a solution for us to review. The reason I went with this approach was because it was easier to make it 'feature complete'.

If we allow saving a model w/o annotation, that annotation-less model now needs to be included in the return of `list_model`. Solving that will make the logic for `list_model` more complex/messier (w/o listing the first annotation twice).